contracts/registries/PoolRegistry.sol

```diff
@@ -0,0 +1,350 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "../interfaces/ISwap.sol";
+import "../interfaces/IMetaSwap.sol";
+import "../meta/MetaSwapDeposit.sol";
+
+/**
+ * @title PoolRegistry
+ * @notice This contract holds list of pools deployed.
+ */
+contract PoolRegistry is Ownable, AccessControl {
```

Comment:
Want any batch functions in here?

Reply:
Also would this be the place to add functions to support coingecko's currency pair needs? eg something like a `calculateSwapRate(ercA, ercB)` or something? I think having a registry unlocks a lot of stuff, and tighter integration with aggs and data providers (CG, TT, DL etc) is a big one